uarray/tests/test_uarray.py

```diff
@@ -427,3 +427,113 @@ def assert_backend_active(backend):
 
     ua.register_backend(be)
     assert_backend_active(be)
+
+
+def test_determine_backend(nullary_mm):
+    class TypeA:
+        pass
+
+    class TypeB:
+        pass
+
+    mark = "determine_backend_test"
+
+    class TypeBackend:
+        __ua_domain__ = "ua_tests"
+
+        def __init__(self, my_type):
+            self.my_type = my_type
+
+        def __ua_convert__(self, dispatchables, coerce):
+            if not all(
+                type(d.value) is self.my_type and d.type is mark for d in dispatchables
+            ):
+                return NotImplemented
+            return tuple(d.value for d in dispatchables)
+
+        def __ua_function__(self, func, args, kwargs):
+            return self.my_type
+
+    BackendA = TypeBackend(TypeA)
+    BackendB = TypeBackend(TypeB)
+
+    with ua.set_backend(BackendA), pytest.raises(ua.BackendNotImplementedError):
+        with ua.determine_backend(TypeB(), mark, domain="ua_tests"):
+            pass
+
+    with ua.set_backend(BackendA), ua.set_backend(BackendB):
+        with ua.determine_backend(TypeA(), mark, domain="ua_tests"):
+            assert nullary_mm() is TypeA
+
+        with ua.determine_backend(TypeB(), mark, domain="ua_tests"):
+            assert nullary_mm() is TypeB
+
+    BackendAny = DisableBackend()  # Has no __ua_convert__, so accepts anything
+    with ua.set_backend(BackendAny):
+        with ua.determine_backend(TypeB(), mark, domain="ua_tests"):
+            assert nullary_mm() is BackendAny.ret
```

Comment:
I'd change this to raise an error, if possible.

Reply:
Done